android/src/main/java/com/reactnativestripesdk/pushprovisioning/AddToWalletButtonView.kt

```diff
@@ -0,0 +1,152 @@
+import com.facebook.react.uimanager.ThemedReactContext
+import com.facebook.react.uimanager.UIManagerModule
+import com.facebook.react.uimanager.events.EventDispatcher
+import com.reactnativestripesdk.PushProvisioningProxy
+import com.reactnativestripesdk.createError
+
+
+class AddToWalletButtonView(private val context: ThemedReactContext, private val requestManager: RequestManager) : AppCompatImageView(context) {
+  private var cardDescription: String? = null
+  private var cardLastFour: String? = null
+  private var ephemeralKey: String? = null
+  private var sourceMap: ReadableMap? = null
+  private var token: ReadableMap? = null
+
+  private var mEventDispatcher: EventDispatcher? = context.getNativeModule(UIManagerModule::class.java)?.eventDispatcher
+  private var loadedSource: GlideUrl? = null
+  private var heightOverride: Int = 0
+  private var widthOverride: Int = 0
+
+  override fun performClick(): Boolean {
+    super.performClick()
+    cardDescription?.let { cardDescription ->
+      ephemeralKey?.let { ephemeralKey ->
+        PushProvisioningProxy.invoke(
+          context.reactApplicationContext,
+          this,
+          cardDescription,
+          ephemeralKey,
+          token)
+      } ?: {
+        dispatchEvent(
+          createError("Failed", "Missing parameters. `ephemeralKey` must be supplied in the props to <AddToWalletButton />")
+        )
+      }
+    } ?: run {
+      dispatchEvent(
+        createError("Failed", "Missing parameters. `cardDescription` must be supplied in the props to <AddToWalletButton />")
+      )
+    }
+    return true
+  }
+
+  init {
+    this.setOnTouchListener { view, event ->
+      if (event.action == MotionEvent.ACTION_DOWN) {
+        view.performClick()
+        return@setOnTouchListener true
+      }
+      return@setOnTouchListener false
+    }
+  }
+
+  fun onAfterUpdateTransaction() {
+    val sourceToLoad = createUrlFromSourceMap(sourceMap)
+    if (sourceToLoad == null) {
+      requestManager.clear(this)
+      setImageDrawable(null)
+      loadedSource = null
+    } else if (sourceToLoad != loadedSource || (heightOverride > 0 || widthOverride > 0)) {
+      loadedSource = sourceToLoad
+      val scale = sourceMap?.getDouble("scale") ?: 1.0
+
+      requestManager
+        .asDrawable()
+        .load(sourceToLoad)
+        .addListener(object : RequestListener<Drawable> {
+          override fun onLoadFailed(e: GlideException?, model: Any?, target: Target<Drawable>?, isFirstResource: Boolean): Boolean {
+            dispatchEvent(
+              createError("Failed", "Failed to load the source from $sourceToLoad")
+            )
+            return true
+          }
+          override fun onResourceReady(resource: Drawable?, model: Any?, target: Target<Drawable>?, dataSource: DataSource?, isFirstResource: Boolean): Boolean {
+            setImageDrawable(
+              RippleDrawable(
+                ColorStateList.valueOf(Color.parseColor("#e0e0e0")),
```

Comment:
Is this color Google pay brand guideline compliant?

Reply:
I used the same color code that is used by default in the google pay buttons, but couldn't find any branding guidelines about ripple color [here](https://developers.google.com/pay/issuers/apis/push-provisioning/android/branding-guidelines#flutter)